### DISCUSSION TITLE
worker creation is failed because of content security policy issue.

### DISCUSSION THREAD

**Discussion by jisnajc** [perspective-dev/perspective#2027] [Q&A]
I am trying to load the perspective table in angular app with the sample url https://xxx/yy. The perspective worker is creating in the base url http://xxx and throwing the content security policy issues.

<img width="944" alt="Capture" src="https://user-images.githubusercontent.com/68993854/205317745-63166e6b-7017-4779-917b-d13bb0b1739b.PNG">

Is this a problem with creating the worker in the base path? If so, how can we override the worker creation in a specified path, say https://xxx/yy.
Also tried to set the custom request header like this. "script-src 'self' https://unpkg.com/@finos/perspective-viewer-datagrid 'unsafe-eval' 'unsafe-inline'; object-src 'self'; worker-src 'self' https://unpkg.com/@finos/perspective-viewer-datagrid blob:;" Then the error disappeared but still the table is not loading. Is there any work around for this issue?

**Comment by timkpaine** [comment]
Your server is responsible for specifying its content policies when it serves your app. You will need to set the appropriate headers (as you have done). This is not a bug or anything, just a detail of how browsers work. 

Now that you have no errors loading your assets, see if the table is present but 0 height or width.

**Reply by jisnajc** [reply]
Definitely table width and height is not the problem. Am getting the error 
'WebSocket is already in CLOSING or CLOSED state.' How can I tackle this?

**Reply by texodus** [reply]
@jisnajc Help us help you.  Your issue is with how you are building and deploying your app - please show us how you got there and specifically how this diverges from the [working examples in the repo](https://github.com/finos/perspective/tree/master/examples).
